pkg/openstack-cinder/config/configsync.go

```diff
@@ -42,22 +61,29 @@ const (
 	infrastructureResourceName = "cluster"
 )
 
-type configSource struct {
-	Namespace string
-	Name      string
-	Key       string
-}
-
 // NewConfigSyncController creates a new instance of ConfigSyncController
-func NewConfigSyncController(c *clients.Clients) factory.Controller {
-	configMapInformer := c.GetConfigMapInformer(util.OpenShiftConfigNamespace)
+func NewConfigSyncController(c *clients.Clients, isHypershift bool) (factory.Controller, error) {
+	var configMapInformer coreinformers.ConfigMapInformer
+
+	if !isHypershift {
+		// If we're in a standalone deployment, we watch the openshift-config namespace
+		configMapInformer = c.GetConfigMapInformer(util.OpenShiftConfigNamespace)
+	} else {
+		// If we're in a hypershift deployment, we watch the clusters-${NAME} namespace
+		// (on the control plane cluster)
+		configMapInformer = c.GetControlPlaneConfigMapInformer(c.ControlPlaneNamespace)
+	}
+
 	controller := &ConfigSyncController{
-		operatorClient:       c.OperatorClient,
-		kubeClient:           c.KubeClient,
-		configMapLister:      configMapInformer.Lister(),
-		infrastructureLister: c.GetInfraInformer().Lister(),
-		namespace:            c.GuestNamespace,
-		eventRecorder:        c.EventRecorder.WithComponentSuffix("ConfigSync"),
+		operatorClient:         c.OperatorClient,
+		configMapLister:        configMapInformer.Lister(),
+		infrastructureLister:   c.GetInfraInformer().Lister(),
+		controlPlaneKubeClient: c.ControlPlaneKubeClient,
+		guestKubeClient:        c.ControlPlaneKubeClient,
```

Comment:
Shouldn't it be `c.KubeClient` ?

Reply:
It should, yes. I reworked this yesterday and must have fat-fingered this.